scio-core/src/main/scala/com/spotify/scio/values/SideInput.scala

```diff
@@ -81,3 +81,35 @@ class SideInputContext[T] private[scio] (val context: DoFn[T, AnyRef]#ProcessCon
   /** Extract the value of a given [[SideInput]]. */
   def apply[S](side: SideInput[S]): S = side.getCache(context)
 }
+
+// Immutable wrapper for j.u.Map[A, B] because .asScala returns a s.c.mutable.Map
+private class SideInputMap[A, B](self: JMap[A, B]) extends Map[A, B] {
+
+  // make eager copies when necessary
+  // scalastyle:off method.name
```

Comment:
Sure. But was thinking if that warning should be added only in the methods that make copies (and only if they happen to be invoked), just in case. But this is minor point anyway.